### ISSUE TITLE
Bug Play/Pause on wireless headphones

### ISSUE THREAD

**Issue by 4e576rt8uh9ij9okp** [futo-org/grayjay-android#685]
I'm using e/OS (Android) and sometimes the play/pause doesn't work.
Playing next and previous videos in a playlist works.

After restarting the app it works again.
I listen to podcasts at work and it's very annoying when I can't pause when someone wants to talk to me.

**Comment by Kelvin-FUTO** [comment]
This issue might correlate with another issue that was fixed in upcoming version 278+ (released tomorrow).

**Comment by 4e576rt8uh9ij9okp** [comment]
> This issue might correlate with another issue that was fixed in upcoming version 278+ (released tomorrow).

The issue is still there, also there is an issue with screen rotation (i will open a new issue for that)

**Comment by 4e576rt8uh9ij9okp** [comment]
Its so annoying to the point that im considering paying youtube even tho I hate YT.

The video stops randomly and i need to close and open the app, these issues didnt exist before.

When i click pause it takes 1-2s to stop then it doesnt then i need to get my phone, unlock, pause... Listen to the person ask me what my fav color is... Then go back 10s and listen again.

Sorry but this makes me so angry... I dont program in kotlin, because i would have done it weeks ago.

**Comment by Blackilykat** [comment]
I have also been experiencing this, I'm on stock pixel 7 pro android with Google buds pro

**Comment by Equinox3141** [comment]
I'm experiencing this as well. Pause needs two taps, that are just far enough apart to not skip to the next video. Play works fine with a single tap. 

I'm running Pixel 6 Pro with android 14, Pixel buds pro, and Grayjay version 264

**Comment by danielkrysiak** [comment]
I'm highly motivated to get this fixed, please let me know if I could help in any way, but I don't know Java and Android SDK at all.

I'm still trying to find anything. I have some additions to what I found in #1577 :

 - I started testing on Samsung S23 Ultra with stock OS, with Shokz OpenRun Pro 2 (the ones which have the issue on Pixel phones with GrapheneOS). I could not reproduce the problem there.
 - On Linux, I looked into `evtest /dev/input/...` of headphones that work fine (Jabra), and the broken ones (Shokz). The output is exactly the same with one difference: Jabra will retry the event multiple times before the sound actually stops.
 - I looked into [Android documentation](https://developer.android.com/media/legacy/media-buttons). I don't think this is the cause, but still interesting: Looks like Grayjay is missing MediaButtonReceiver event handler. Newpipe has it.

**Comment by Zvonimir-FUTO** [comment]
@danielkrysiak could you provide a bit more information on replication steps? Does the issue occur all the time? Is it present only after some period of inactivity? With locked/unlocked screen?

**Comment by danielkrysiak** [comment]
@Zvonimir-FUTO unfortunately I've been holding off on creating this issue for 4-6 months, because I was waiting to notice some pattern. I think I use Grayjay in most ways possible: many hours a day, offline/online, foreground/background, background for hours on hikes - I still have nothing.

I'm going to start taking notes when it breaks.

In the meantime, if I could suggest anything: could you please compare how your implementation could be different from other Open Source project, like Newpipe, Ultrasonic, etc. which work just fine?

**Comment by Zvonimir-FUTO** [comment]
We made a possible fix for this issue by implementing MediaButtonReceiver. Since we cannot reproduce the issue (due to our events being handled by MediaSession instead) we hope it will solve your issue next release. Please let us know. We are also very motivated to have this solved but due to the difficulty in reproducing it has ended up low priority.

**Comment by danielkrysiak** [comment]
@Zvonimir-FUTO is it possible to get access to the dev/nightly builds? I'm not new to this, even if it'll be crashing, corrupting data, etc. I'm fine with that. Likewise, I could help with other testing as well if you like.

**Comment by Zvonimir-FUTO** [comment]
> @Zvonimir-FUTO is it possible to get access to the dev/nightly builds? I'm not new to this, even if it'll be crashing, corrupting data, etc. I'm fine with that. Likewise, I could help with other testing as well if you like.

Currently we have no nightly builds. You can download the unstable version of Grayjay from the following link:

https://releases.grayjay.app/app-universal-release-unstable.apk

Although keep in mind that there is currently no way to track unstable versions; it doesn't update from the app, nor do we notify when a new version is released. Another thing is the way versions change—every release gets its own version, including unstable, so stable versions sometimes skip few version numbers used for unstable versions. Also stable versions can be newer versions with latest changes.

We currently release unstable versions internally to test before we release stable ones, which are released incrementally.

**Comment by Zvonimir-FUTO** [comment]
Can you guys test again with v276 and report back if the issue is still present?

**Comment by danielkrysiak** [comment]
I had troubles reproducing for the last 3 days on 268, only happened 2 times, but what I can confirm so far, only restarting Grayjay fixes it. Cycling headphones or Bluetooth doesn't.
I'm switching to 276 now.

**Comment by danielkrysiak** [comment]
I'll be 100% sure in a couple of days, but I'm 95% sure at this point: it's fixed.
More than that, play/pause works a lot quicker than in previous versions, and as expected: resuming after a long pause started working as well.

Sorry for an off-topic, but while testing thoroughly lately, I've noticed some regressions since 265, I'll do separate tickets when I collect more data, but I think it's worth mentioning them anyway:
 - Downloads worked perfectly on Grayjay, since 265 they often get stuck.
 - Resume on network reconnection worked perfectly on Grayjay, since 265 it often requires loading video again.
 -  Battery drain, looks like Grayjay is always active in background even if not playing anything (battery usage: optimized).

**Comment by Zvonimir-FUTO** [comment]
Closing as solved.

**Comment by Hotsauwz** [comment]
I would like to note that im also having this issue with the spotify plugin using both my JBL tune 770NC and my JBL tune 230NC
